libtsuba/src/RDGPartHeader.cpp

```diff
@@ -316,12 +338,54 @@ katana::from_json(const json& j, katana::RDGPartHeader& header) {
   j.at(kPartPropertyFilesKey).get_to(header.part_prop_info_list_);
   j.at(kPartPropertyMetaKey).get_to(header.metadata_);
 
+  /// Storage Format Version Handling
+
+  // support loading "storage_format_version=1" RDGs, aka RDGs without
+  // an explicit storage_format_version
   if (auto it = j.find(kStorageFormatVersionKey); it != j.end()) {
     it->get_to(header.storage_format_version_);
   } else {
     header.storage_format_version_ = kPartitionStorageFormatVersion1;
   }
 
+  // load the unstable_storage_format flag if it is present in the RDG
+  // RDGs created before support for unstable_storage_format was added do not have this flag.
+  // If it is not present, the RDG is assumed to *NOT* be stored in an unstable_storage_format since the rdg was created before unstable formats were introduced
+  if (auto it = j.find(kUnstableStorageFormatFlagKey); it != j.end()) {
+    it->get_to(header.unstable_storage_format_);
+    // Ensure unstable_storage_format RDGs are not loaded when the feature flag is not set
+  } else {
+    header.unstable_storage_format_ = false;
+  }
+
+  if (KATANA_EXPERIMENTAL_ENABLED(UnstableRDGStorageFormat)) {
+    KATANA_LOG_WARN(
+        "UnstableRDGStorageFormat is set. RDGs will be stored in an unstable "
+        "storage format. Loading RDGs stored in an unstable storage format "
+        "will be permitted.");
+    if (header.unstable_storage_format_) {
+      KATANA_LOG_WARN(
+          "Loading an RDG stored in an unstable storage format. If any issues "
+          "are encountered, please regenerate this RDG before filing any bugs, "
+          "as the unstable storage format can change without notice.");
+    }
+  } else {
+    if (header.unstable_storage_format_) {
+      throw std::runtime_error(
```

Comment:
Is there any reason why we are throwing runtime errors here instead of following the other pattern of returning a `Result<T>`?

Reply:
Great question. From what I understand the way the `nlohmann` `to_json` and `from_json` functions have to be void? I can't see this specified in any of the docs specifically, but every example uses void. 
I would love to be proven wrong here though.

Reply:
This is right; our wrappers for `nlohmann`'s functions catch these exceptions and turn them into `Result<T>`'s

Reply:
Gotcha, thanks!